### DISCUSSION TITLE
VBox is not working in 3.4.0rc1

### DISCUSSION THREAD

**Discussion by GillesFa** [bokeh/bokeh#13742] [General]
Hi,
I have install the following bokeh version:

Python version        :  3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:26:23) [MSC v.1916 64 bit (AMD64)]
IPython version       :  8.17.2
Tornado version       :  6.3.3
Bokeh version         :  3.4.0rc1
BokehJS static path   :  C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\server\static
node.js version       :  (not installed)
npm version           :  (not installed)
jupyter_bokeh version :  (not installed)
Operating system      :  Windows-10-10.0.19045-SP0

I am trying to use a VBox layout but when I run my attached example, I get the following error:

> Traceback (most recent call last):
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\runpy.py", line 198, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\runpy.py", line 88, in _run_code
>     exec(code, run_globals)
>   File "c:\Users\gilles.faure\.vscode\extensions\ms-python.python-2024.0.1\pythonFiles\lib\python\debugpy\adapter/../..\debugpy\launcher/../..\debugpy\__main__.py", line 39, in <module>
>     cli.main()
>   File "c:\Users\gilles.faure\.vscode\extensions\ms-python.python-2024.0.1\pythonFiles\lib\python\debugpy\adapter/../..\debugpy\launcher/../..\debugpy/..\debugpy\server\cli.py", line 430, in main
>     run()
>   File "c:\Users\gilles.faure\.vscode\extensions\ms-python.python-2024.0.1\pythonFiles\lib\python\debugpy\adapter/../..\debugpy\launcher/../..\debugpy/..\debugpy\server\cli.py", line 284, in run_file
>     runpy.run_path(target, run_name="__main__")
>   File "c:\Users\gilles.faure\.vscode\extensions\ms-python.python-2024.0.1\pythonFiles\lib\python\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 321, in run_path
>     return _run_module_code(code, init_globals, run_name,
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "c:\Users\gilles.faure\.vscode\extensions\ms-python.python-2024.0.1\pythonFiles\lib\python\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 135, in _run_module_code
>     _run_code(code, mod_globals, init_globals,
>   File "c:\Users\gilles.faure\.vscode\extensions\ms-python.python-2024.0.1\pythonFiles\lib\python\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 124, in _run_code
>     exec(code, run_globals)
>   File "C:\Development\Net-Zero_Energy_Networks\energy_optimisation\scratch\bokeh_vbox_issue.py", line 9, in <module>
>     show(vbox)
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\io\showing.py", line 149, in show
>     return _show_with_state(obj, state, browser, new, notebook_handle=notebook_handle)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\io\showing.py", line 200, in _show_with_state
>     _show_file_with_state(obj, state, new, controller)
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\io\showing.py", line 181, in _show_file_with_state
>     filename = save(obj, state=state)
>                ^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\io\saving.py", line 98, in save
>     _save_helper(obj, filename, resources, title, template, theme)
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\io\saving.py", line 164, in _save_helper
>     html = file_html(obj, resources=resources, title=title, template=template or FILE, theme=theme)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\embed\standalone.py", line 358, in file_html
>     with OutputDocumentFor(models_seq, apply_theme=theme, always_new=_always_new) as doc:
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\contextlib.py", line 137, in __enter__
>     return next(self.gen)
>            ^^^^^^^^^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\embed\util.py", line 175, in OutputDocumentFor
>     doc.validate()
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\document\document.py", line 787, in validate
>     issues = check_integrity(refs)
>              ^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\core\validation\check.py", line 171, in check_integrity
>     issues.error.extend(func())
>                         ^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\core\validation\decorators.py", line 74, in _wrapper
>     extra = func(*args, **kwargs)
>             ^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\models\layouts.py", line 458, in _check_repeated_layout_children
>     children = [ item["child"] for item in self.items ]
>                                            ^^^^^^^^^^
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\core\has_props.py", line 371, in __getattr__
>     self._raise_attribute_error_with_matches(name, properties)
>   File "C:\Users\gilles.faure\AppData\Local\miniconda3\envs\EO_py3.11\Lib\site-packages\bokeh\core\has_props.py", line 379, in _raise_attribute_error_with_matches
>     raise AttributeError(f"unexpected attribute {name!r} to {self.__class__.__name__}, {text} attributes are {nice_join(matches)}")
> AttributeError: unexpected attribute 'items' to VBox, possible attributes are align, aspect_ratio, children, context_menu, css_classes, css_variables, disabled, elements, flow_mode, height, height_policy, js_event_callbacks, js_property_callbacks, margin, max_height, max_width, min_height, min_width, name, resizable, rows, sizing_mode, spacing, styles, stylesheets, subscribed_events, syncable, tags, visible, width or width_policy

Could you advise what is the issue with my code?
thanks
Gilles

ps: couldn't attached the py file so here the code:

```
from bokeh.plotting import show
from bokeh.models import TextInput
from bokeh.models import VBox



text_input = TextInput(value="default", title="Label:")
vbox = VBox(children=[text_input])
show(vbox)


```

**Comment by bryevdv** [comment]
@GillesFa The error tries to be self-explanatory:

> AttributeError: unexpected attribute 'items' to VBox

Perhaps you mean to set the `children` property, which does exist?

https://github.com/bokeh/bokeh/blob/8a5a35eb078e386c14b5580a51e8a6673d57d197/src/bokeh/models/layouts.py#L444-L446

**Reply by GillesFa** [reply]
isn't it what I am doing?
vbox = VBox(**children**=[text_input])

**Comment by bryevdv** [comment]
My apologies, indeed you are. That's a bug in the validator:

https://github.com/bokeh/bokeh/blob/8a5a35eb078e386c14b5580a51e8a6673d57d197/src/bokeh/models/layouts.py#L456-L460
